android/thymiovpl/src/org/mobsya/thymiovpl/DashelSerial.java

```diff
@@ -1,150 +0,0 @@
-        QtActivity.getQtActivityInstance().registerReceiver(mUsbReceiver, filter);
-        manager.requestPermission(device, mPermissionIntent);
-        latch.await();
-    }
-
-    public DashelSerial() throws InterruptedException {
-        manager = (UsbManager) QtActivity.getQtActivityInstance().getSystemService(Context.USB_SERVICE);
-        device = (UsbDevice)  QtActivity.getQtActivityInstance().getIntent().getParcelableExtra(UsbManager.EXTRA_DEVICE);
-        if(device == null) {
-            HashMap<String, UsbDevice> deviceList = manager.getDeviceList();
-            Iterator<UsbDevice> deviceIterator = deviceList.values().iterator();
-            while(deviceIterator.hasNext()){
-                device = deviceIterator.next();
-                if(device.getVendorId() == 0x0617 || device.getProductId() == 0x000a) {
-                    RequestPermission();
-                    break;
-                }
-            }
-            Log.i("DashelJ", "Unable to find usb device");
-            return;
-        }
-    }
-
-    int OpenDevice()
-    {
-        if (device == null)
-            return -1;
-
-        connection = manager.openDevice(device);
-
-        if(connection == null) {
-            Log.i("DashelJ", "Unable to open device");
-            return -1;
-        }
-
-        int i;
-        // Claim interface 0 (control)
-        UsbInterface intf = device.getInterface(0);
-        connection.claimInterface(intf, true);
-
-        // Look for the cdc-acm data interface
-        for(i = 0; i < device.getInterfaceCount(); i++) {
-            intf = device.getInterface(i);
-            if(intf.getInterfaceClass() == UsbConstants.USB_CLASS_CDC_DATA) {
-                // Claim it, and look for Bulk IN and Bulk OUT endpoints
-                connection.claimInterface(intf, true);
-                int j;
-                for(j = 0; j < intf.getEndpointCount() && (epin == -1 || epout == -1); j++) {
-                    UsbEndpoint ep = intf.getEndpoint(j);
-                    if(ep.getType() == UsbConstants.USB_ENDPOINT_XFER_BULK) {
-                        if(epin == -1 && ep.getDirection() == UsbConstants.USB_DIR_IN)
-                            epin = ep.getAddress();
-                        if(epout == -1 && ep.getDirection() == UsbConstants.USB_DIR_OUT)
-                            epout = ep.getAddress();
-                    }
-                }
-                break;
-            }
-        }
-
-        if(epin == -1 || epout == -1) {
-            Log.i("DashelJ", "Unable to get data endpoints");
-            return -1;
-        }
-
-        Log.i("DashelJ", "Device successfully opened");
-        return connection.getFileDescriptor();
-    }
-
-    int GetEpIn()
-    {
-        return epin;
-    }
-
-    int GetEpOut()
-    {
-        return epout;
-    }
-
-    boolean CloseDevice()
-    {
-        connection.close(); // Linux take care of releasing everything else ...
-        return true;
-    }
-}
```

Comment:
I just realised that thymio vpl 2 might be using this file (and the rest of Dashel android), we should move it somewhere not delete it. I will check later.

Reply:
It's ok, VPL2 has a modified copy of this file: https://github.com/aseba-community/thymio-vpl2/blob/master/android/src/org/mobsya/thymiovpl/DashelSerial.java

Reply:
thanks @marvelous, so we can delete it here.